README.md

```diff
@@ -91,14 +91,21 @@ The following events trigger the publishing service:
 
 The event publishing SPI is defined as an event listener in the `build.conf` as `spi-events-listener-event-publisher-enabled=true`
 
+## Database Credentials Service (SPI)
+
+This is an SPI which is provided by Quarkus which is the underlying framework which Keycloak runs on. The configuration is set in `quarkus.properties`
+
+There is a single class, `RDSCredentialsProvider` with a `getCredentials` method. 
+This gets the password using the `generateAuthenticationToken` on `RDSUtilities` and returns it which is then used to connect to the database. 
+There is a cache which refreshes every 5 minutes so we're not calling AWS for every new database connection.
 
 ## Updating TDR Realm Configuration json
 
 A separate GitHub actions workflow is used to update the TDR realm configuration: [TDR Update Auth Server Configuration](https://github.com/nationalarchives/tdr-auth-server/actions/workflows/update.yml)
 
 This is because on container start up, if a realm already exists, its configuration json is ignored. So restarting the container will not update the configuration.
 
-The TDR Auth Server Update Jenkins job updates existing realm configuration using the Keycloak REST APIs: https://www.keycloak.org/docs-api/11.0/rest-api/index.html
+The TDR Update Auth Server Configuration GitHub actions workflow updates existing realm configuration using the Keycloak REST APIs: https://www.keycloak.org/docs-api/11.0/rest-api/index.html
```

Comment:
The Keycloak API docs link needs updating, its pointing to version 11

Reply:
Ah yeah, I've updated it to 18.0. They don't seem to do a `latest` url so we'll have to keep this up to date next time we do a major version upgrade.